### PR TITLE
Fix query module addr

### PR DESCRIPTION
**Pull request by piobab**
Address handler will be improved with this task https://delphilabs.atlassian.net/browse/MP-2079 so for now solved with addresses separation.

### CODE REVIEW

File: packages/types/src/address_provider.rs [hunk 1]
```diff
@@ -146,28 +146,43 @@ pub mod helpers {
...
+    /// It fails if the provided address does not start with current chain prefix.
+    pub fn query_contract_addr(
         deps: Deps<impl cosmwasm_std::CustomQuery>,
         address_provider_addr: &Addr,
         contract: MarsAddressType,
     ) -> StdResult<Addr> {
         let res: AddressResponseItem =
             deps.querier.query_wasm_smart(address_provider_addr, &QueryMsg::Address(contract))?;
-
         deps.api.addr_validate(&res.address)
```
Comment (larry0x) [inline packages/types/src/address_provider.rs:159]: Imo this syntax is cleaner

```suggestion
        deps.querier
            .query_wasm_smart::<AddressResponseItem>(
                address_provider_addr,
                &QueryMsg::Address(contract),
            )
            .map(|res| deps.api.addr_validate(&res.address))?
```

File: packages/types/src/address_provider.rs [hunk 2]
```diff
@@ -146,28 +146,43 @@ pub mod helpers {
...
         address_provider_addr: &Addr,
         contracts: Vec<MarsAddressType>,
     ) -> StdResult<HashMap<MarsAddressType, Addr>> {
         let res: Vec<AddressResponseItem> = deps
             .querier
             .query_wasm_smart(address_provider_addr, &QueryMsg::Addresses(contracts))?;
-
         res.iter()
             .map(|item| Ok((item.address_type, deps.api.addr_validate(&item.address)?)))
             .collect()
```
Comment (larry0x) [inline packages/types/src/address_provider.rs:175]:
```suggestion
        deps.querier
            .query_wasm_smart::<Vec<AddressResponseItem>>(
                address_provider_addr,
                &QueryMsg::Addresses(contracts),
            )?
            .into_iter()
            .map(|item| Ok((item.address_type, deps.api.addr_validate(&item.address)?)))
            .collect()
```

File: packages/types/src/address_provider.rs [hunk 3]
```diff
@@ -146,28 +146,43 @@ pub mod helpers {
...
+
+    /// Query Mars Hub module address
+    pub fn query_module_addr(
+        deps: Deps<impl cosmwasm_std::CustomQuery>,
+        address_provider_addr: &Addr,
+        module: MarsAddressType,
+    ) -> StdResult<String> {
+        let res: AddressResponseItem =
+            deps.querier.query_wasm_smart(address_provider_addr, &QueryMsg::Address(module))?;
+        Ok(res.address)
```
Comment (larry0x) [inline packages/types/src/address_provider.rs:186]:
```suggestion
        deps.querier
            .query_wasm_smart::<AddressResponseItem>(
                address_provider_addr,
                &QueryMsg::Address(module),
            )
            .map(|res| res.address)
```

File: packages/types/src/address_provider.rs [hunk 4]
```diff
@@ -146,28 +146,43 @@ pub mod helpers {
...
             deps.querier.query_wasm_smart(address_provider_addr, &QueryMsg::Address(contract))?;
-
         deps.api.addr_validate(&res.address)
     }
 
-    pub fn query_addresses(
+    /// Query contract addresses.
+    ///
+    /// It fails if the provided address does not start with current chain prefix.
+    pub fn query_contracts_addr(
```
Comment (larry0x) [inline packages/types/src/address_provider.rs:165]: Nitpick: the correct grammar is `query_contract_addrs`

**Review by larry0x — CHANGES_REQUESTED** [review]

**Comment by larry0x** [comment]
Unrelated note: these queries can be raw queries to save gas. We can do this later when moving to the adapter approach.

File: packages/types/src/address_provider.rs (hunk 1 above)
Comment (piobab) [reply packages/types/src/address_provider.rs]: ok ;)

Comment (piobab) [reply packages/types/src/address_provider.rs]: ```Applying suggestions on deleted lines is currently not supported. Please  and try again.```

Comment (larry0x) [reply packages/types/src/address_provider.rs]: @piobab I pushed a commit

**Review by brimigs — APPROVED** [review]

**Pull request merged**